src/commands/staff/sudoreply.ts

```diff
@@ -1,45 +1,104 @@
 import { PVQuestion } from '@lib/types/PVQuestion';
-import { BOT, DB, PREFIX } from '@root/config';
-import { staffPerms } from '@lib/permissions';
-import { Message, MessageEmbed } from 'discord.js';
-import { Command } from '@root/src/lib/types/Command';
+import { BOT, DB, MAINTAINERS } from '@root/config';
+import { ADMIN_PERMS, staffPerms, STAFF_PERMS } from '@lib/permissions';
+import { ApplicationCommandOptionData, ApplicationCommandPermissionData, CommandInteraction, GuildChannel, Message, MessageEmbed, TextChannel, ThreadChannel } from 'discord.js';
+import { Command } from '@lib/types/Command';
+import { Course } from '@lib/types/Course';
 
 export default class extends Command {
 
 	description = `Reply to a question asked through ${BOT.NAME}.`;
 	usage = '<questionID> <response>';
-	extendedHelp = 'Responses get sent to the askers DMs. This command will tell you it failed if it cannot send the DM.';
+	extendedHelp = 'Responses are put into a private thread between you and the asker.';
 	runInDM = false;
-	aliases = ['sreply'];
 
-	permissions(msg: Message): boolean {
-		return staffPerms(msg);
-	}
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'questionid',
+			description: 'ID of question you are replying to',
+			type: 'INTEGER',
```

Comment:
The function I wrote to generate questions IDs returns stings that can have leading zeros. These questions could never be reached with an argument type of integer.